File: scripts/verify-lp-hero.py
import urllib.request
import json
import sys

DIRECTUS_URL = 'http://localhost:8055'
ADMIN_EMAIL = 'admin@spead.ai'
ADMIN_PASSWORD = 'password123'

def request(method, endpoint, data=None, token=None):
    url = f"{DIRECTUS_URL}{endpoint}"
    headers = {'Content-Type': 'application/json'}
    if token:
        headers['Authorization'] = f"Bearer {token}"
    body = json.dumps(data).encode('utf-8') if data else None
    
    try:
        req = urllib.request.Request(url, data=body, headers=headers, method=method)
        with urllib.request.urlopen(req) as response:
            return json.loads(response.read().decode('utf-8'))
    except urllib.request.HTTPError as e:
        return {'error': True, 'status': e.code, 'body': json.loads(e.read().decode('utf-8'))}

# Login
login = request('POST', '/auth/login', {'email': ADMIN_EMAIL, 'password': ADMIN_PASSWORD})
token = login['data']['access_token']

# Check Fields
print("🔍 Checking Fields for lp_hero...")
res = request('GET', '/fields/lp_hero', None, token)

if 'error' in res:
    print(f"❌ Failed to list fields: {res}")
    sys.exit(1)

fields = res['data']
existing = [f['field'] for f in fields]

required = ['badge_text', 'div_main_copy', 'div_cta', 'div_visuals', 'hero_visual']
missing = [r for r in required if r not in existing]

print(f"{'Field':<25} | {'Type':<10} | {'Sort':<5}")
print("-" * 50)
for f in sorted(fields, key=lambda x: x['meta']['sort'] if x['meta'] and x['meta']['sort'] else 999):
    sort = f['meta']['sort'] if f['meta'] else 'N/A'
    print(f"{f['field']:<25} | {f['type']:<10} | {sort:<5}")

if missing:
    print(f"\n❌ MISSING FIELDS: {', '.join(missing)}")
    sys.exit(1)
else:
    print("\n✅ ALL REQUIRED FIELDS PRESENT.")

File: scripts/verify-lp-hero.js
import dotenv from 'dotenv';
// dotenv.config();

const DIRECTUS_URL = 'http://localhost:8055';
const ADMIN_EMAIL = 'admin@spead.ai';
const ADMIN_PASSWORD = 'password123';

async function verifyLpHero() {
    try {
        console.log(`🔌 Connecting to Directus at ${DIRECTUS_URL}...`);

        // 1. Authenticate
        const loginRes = await fetch(`${DIRECTUS_URL}/auth/login`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ email: ADMIN_EMAIL, password: ADMIN_PASSWORD })
        });

        if (!loginRes.ok) throw new Error(`Login failed: ${loginRes.status}`);
        const access_token = (await loginRes.json()).data.access_token;
        const headers = {
            'Authorization': `Bearer ${access_token}`,
            'Content-Type': 'application/json'
        };
        console.log('✅ Authenticated.');

        // 2. List Fields
        console.log('\n🔍 Listing Fields for lp_hero...');
        const res = await fetch(`${DIRECTUS_URL}/fields/lp_hero`, { headers });
        const data = await res.json();

        if (!res.ok) {
            console.error('❌ Failed to list fields:', JSON.stringify(data));
            process.exit(1);
        }

        const fields = data.data.map(f => ({
            field: f.field,
            type: f.type,
            interface: f.meta?.interface,
            sort: f.meta?.sort
        })).sort((a, b) => (a.sort || 999) - (b.sort || 999));

        console.table(fields);

        // Check for specific fields
        const missing = ['badge_text', 'div_main_copy', 'div_cta', 'div_visuals', 'hero_visual']
            .filter(req => !fields.find(f => f.field === req));

        if (missing.length > 0) {
            console.error('\n❌ MISSING FIELDS:', missing.join(', '));
            process.exit(1);
        } else {
            console.log('\n✅ ALL REQUIRED FIELDS PRESENT.');
        }

    } catch (err) {
        console.error('\n❌ Script failed:', err.message);
        process.exit(1);
    }
}

verifyLpHero();
